src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -818,6 +818,35 @@ void QgsLayoutItemLegend::onAtlasEnded()
   updateFilterByMap();
 }
 
+QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
+{
+  QgsExpressionContext context = QgsLayoutItem::createExpressionContext();
+
+  //Can't utilize QgsExpressionContextUtils::mapSettingsScope as we don't always
+  //have a QgsMapSettings object available when the context is required, so we manually
+  //add the same variables here
+
+  // add the existing context of the linked map
+  if (mMap)
+  {
+    context.appendScope( mMap.createExpressionContext() );
+  }
+
+  QgsExpressionContextScope *scope = new QgsExpressionContextScope( tr( "Legend Settings" ) );
+
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_title" ),title(),true);
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_settings" ),legendSettings(),true);
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "column_count" ),columnCount(),true);
```

Comment:
can you make sure all these variables are prefixed with "legend_", i.e. "legend_column_count".